### PR TITLE
add detach instance method

### LINKED ISSUE #6315: detach

### PR DESCRIPTION
**Pull request by ahmedo42**
closes #6315

### CODE REVIEW

File: ivy/functional/frontends/torch/Tensor.py [hunk 1]
```diff
@@ -145,6 +145,9 @@ def new_tensor(
     def view_as(self, other):
         return self.view(other.shape)
 
+    def detach(self):
+        return self.new_tensor(self.data, requires_grad=False)
```
Comment (VedPatwardhan) [inline ivy/functional/frontends/torch/Tensor.py:149]: I think the ivy equivalent of torch's `detach` is the `stop_gradient` function, please use it with a `preserve_type=False` in order to get a tensor with `requires_grad=False`.

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hi @ahmedo42, just requested a few changes, rest looks good to me 🙂

File: ivy/functional/frontends/torch/Tensor.py (hunk 1 above)
Comment (VedPatwardhan) [reply ivy/functional/frontends/torch/Tensor.py]: Yeah but I think we could assume that the `torch.detach` function used inside the `stop_gradient` torch implementation should be more efficient with memory that our implementation.

**Review by VedPatwardhan — APPROVED** [review]
lgtm!
